internal/fs/vss_windows.go

```diff
@@ -0,0 +1,833 @@
+		return VssSnapshot{}, fmt.Errorf(
+			"VSS error: IsVolumeSupported() returned %s (%#x)",
+			hRes.Str(), hRes,
+		)
+	} else if !isSupported {
+		return VssSnapshot{}, fmt.Errorf(
+			"VSS error: Snapshots are not supported for volume %s",
+			volume,
+		)
+	}
+
+	// TODO:
+	//   What about IVssBackupComponents::AddComponent?
+	//   see https://docs.microsoft.com/en-us/windows/desktop/VSS/simple-shadow-copy-creation-for-backup
+
+	hRes, snapshotSetID := iVssBackupComponents.StartSnapshotSet()
+	if hRes != S_OK {
+		return VssSnapshot{}, fmt.Errorf(
+			"VSS error: StartSnapshotSet() returned %s (%#x)", hRes.Str(), hRes,
+		)
+	}
+
+	if hRes := iVssBackupComponents.AddToSnapshotSet(volume, &snapshotSetID); hRes != S_OK {
+		return VssSnapshot{}, fmt.Errorf(
+			"VSS error: AddToSnapshotSet() returned %s (%#x)", hRes.Str(), hRes,
+		)
+	}
+
+	err = handleAsyncFunctionCall(
+		iVssBackupComponents.PrepareForBackup,
+		"PrepareForBackup", timeoutInMillis,
+	)
+	if err != nil {
+		return VssSnapshot{}, err
+	}
+
+	err = handleAsyncFunctionCall(
+		iVssBackupComponents.DoSnapshotSet, "DoSnapshotSet", timeoutInMillis,
+	)
+	if err != nil {
+		return VssSnapshot{}, err
+	}
+
+	var snapshotProperties VssSnapshotProperties
+	if hRes := iVssBackupComponents.GetSnapshotProperties(
+		snapshotSetID, &snapshotProperties,
+	); hRes != S_OK {
+		return VssSnapshot{}, fmt.Errorf(
+			"VSS error: GetSnapshotProperties() returned %s (%#x)",
+			hRes.Str(), hRes,
+		)
+	}
+
+	/*
+
+		https://microsoft.public.win32.programmer.kernel.narkive.com/aObDj2dD/volume-shadow-copy-backupcomplete-and-vss-e-bad-state
+
+		CreateVSSBackupComponents();
+		InitializeForBackup();
+		SetBackupState();
+		GatherWriterMetadata();
+		StartSnapshotSet();
+		AddToSnapshotSet();
+		PrepareForBackup();
+		DoSnapshotSet();
+		GetSnapshotProperties();
+		<Backup all files>
+		VssFreeSnapshotProperties();
+		BackupComplete();
+	*/
+
+	return VssSnapshot{
+		iVssBackupComponents, snapshotSetID,
+		snapshotProperties, timeoutInMillis,
+	}, nil
+}
+
+// asyncCallFuncfunc is the callback type for handleAsyncFunctionCall.
+type asyncCallFuncfunc func() (HRESULT, *IVSSAsync)
+
+// handleAsyncFunctionCall calls an async functions and waits for it to either
+// finish or timeout.
+func handleAsyncFunctionCall(
+	function asyncCallFuncfunc, name string, timeoutInMillis int64,
+) error {
+
+	if hRes, iVssAsync := function(); hRes != S_OK {
+		return fmt.Errorf(
+			"VSS error: %s() returned %s (%#x)", name, hRes.Str(), hRes,
+		)
+	} else {
+
+		if iVssAsync == nil {
+			return fmt.Errorf("VSS error: %s() returned nil", name)
+		}
+
+		err := iVssAsync.WaitUntilAsyncFinished(timeoutInMillis)
+		iVssAsync.Release()
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// loadIVssBackupComponentsConstructor finds the constructor of the VSS api
+// inside the VSS dynamic library.
+func loadIVssBackupComponentsConstructor() (*windows.LazyProc, error) {
+
+	createInstanceName :=
```

Comment:
Empty line.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/ea769e845ff18c558c098f265b798918e2a4a28d